djangocms_navigation/cms_config.py

```diff
@@ -1,67 +1,103 @@
+from functools import partial
+
 from django.conf import settings
 from django.core.exceptions import ImproperlyConfigured
 
 from cms.app_base import CMSAppConfig, CMSAppExtension
 from cms.models import Page
 
-from .utils import purge_menu_cache, get_model
+from .utils import purge_menu_cache
 from .constants import MENU_MODEL, ITEM_MODEL
-
-
-MenuContent = get_model(MENU_MODEL)
-MenuItem = get_model(ITEM_MODEL)
+from .models import MenuContent as NavigationMenuContent
+from .models import MenuItem as NavigationMenuItem
 
 
 class NavigationCMSExtension(CMSAppExtension):
     def __init__(self):
         self.navigation_apps_models = {}
 
     def configure_app(self, cms_config):
-        if hasattr(cms_config, "navigation_models"):
-            navigation_app_models = getattr(cms_config, "navigation_models")
-            if isinstance(navigation_app_models, dict):
-                self.navigation_apps_models.update(navigation_app_models)
-            else:
-                raise ImproperlyConfigured(
-                    "navigation configuration must be a dictionary object"
-                )
-        else:
+        if not hasattr(cms_config, "navigation_models"):
             raise ImproperlyConfigured(
                 "cms_config.py must have navigation_models attribute"
             )
 
+        navigation_app_models = getattr(cms_config, "navigation_models")
+        if isinstance(navigation_app_models, dict):
+            self.navigation_apps_models.update(navigation_app_models)
+        else:
+            raise ImproperlyConfigured(
+                "navigation configuration must be a dictionary object"
+            )
+
+        if not hasattr(cms_config, 'djangocms_versioning_enabled'):
+            cms_config.djangocms_versioning_enabled = False
+
+        print(cms_config)
+        print(cms_config.djangocms_versioning_enabled)
+        if cms_config.djangocms_versioning_enabled:
+            from djangocms_versioning.datastructures import VersionableItem
+
+            MenuContent = cms_config.navigation_cms_models[MENU_MODEL]
+            MenuItem = cms_config.navigation_cms_models[ITEM_MODEL]
+            print(MenuContent)
 
-def copy_menu_content(original_content):
+            complete_copy_menu_content = partial(
+                copy_menu_content, MenuContent, MenuItem
+            )
+
+            versioning = [
+                VersionableItem(
+                    content_model=MenuContent,
+                    grouper_field_name="menu",
+                    copy_function=complete_copy_menu_content,
+                    preview_url=MenuContent.get_preview_url,
+                    on_publish=on_menu_content_publish,
+                    on_unpublish=on_menu_content_unpublish,
+                    on_draft_create=on_menu_content_draft_create,
+                    on_archive=on_menu_content_archive,
+                )
+            ]
+            cms_config.versioning = versioning
```

Comment:
Add the following code after line 61

```
            from cms.utils.setup import configure_cms_apps
            versioning_app = apps.get_app_config('djangocms_versioning')
            # will throw an error because configuring versioning apps is not idempotent
            configure_cms_apps([versioning_app])
```